### ISSUE TITLE
FX Send as a per voice parameter

### ISSUE THREAD

**Issue by PopeKehoeIII** [surge-synthesizer/surge#3645]
I'm aiming to create sounds where individual notes can feed the fx send (or not) so that you can do things like send specific notes in a sequence to a delay (or every 4th note played, or whatever) or swell reverb amount with polyAT or envelopes while playing polyphonically. A way to add some polyphonic expression to monophonic fx anyway.

This would require surge having two more voice sum sections (one for each send) rather than just the one - since the HP filter is also a scene parameter currently that would require either two more copies of the simple highpass (with either shared or independent control for each send) as part of the sum block or the high pass would have to become also a voice parameter (which could cause issues with patches sounding different)

Right now there is a workaround but it effectively doubles CPU usage for the synth and uses up scene B - copy scene A into scene B and basically use scene B as your send (modulating per voice gain by whatever you would previously have modulated "send" by) but it's definitely a bit of a nuclear option (duplicating the entire osc/filter/modulation section of the synth when all you really want is another voice sum section)

I tried to find an example in my library of a polysynth capable of doing this and came up blank - it's easy to do monophonically in all kinds of environments (zebrify, bitwig's grid, surge, etc) but despite there being no fundamental logic reason this couldn't be possible I have yet to find an implementation of it (maybe in a more "supersynth" context like omnisphere, but I don't own it)

**Comment by mkruselj** [comment]
I do wonder why is that post-amp highpass per scene rather than calculated per voice, honestly. Can't be that much more expensive to just do it per voice, hence just do the sends and HP per voice too.

**Comment by baconpaul** [comment]
The reason is because the surge algo sums the voices, downsamples them to audio rate from oversample rate, and then hands them to the FX. If you had an FX per voice you would need 3 sceneouts per scene (scene out, scene send 1 out, scene send 2 out) and would need to individually sum and downsample those.

It’s not an easy change or a CPU neutral one as a result.

**Comment by mkruselj** [comment]
Got it. I guess not feasible then!

**Comment by baconpaul** [comment]
It’s feasible; you would do it the way I outlined! It’s just not quick and easy :)

If we had it as a mode that would allow us to do CPU optimization when it was send-per-scene-not-per-voice. But that adds even more complexity (are you voice modulatable or not etc...)

**Comment by mkruselj** [comment]
Yeah I think that's a bit too much, having an option for it. It should be either/or in this case, I feel (especially due the whole scene/voice mod conundrum etc).

**Comment by PopeKehoeIII** [comment]
I realized another issue with doing this as default behavior rather than a switch, especially if the FX routing is not made significantly more flexible (akin to z2's FX grid) - despite the send slider send being in the scene area of the GUI (implying it's sending from the voice sum stage to me) the send fx according to the routing diagram (and in reality having just tested to confirm) is after the insert fx - making per voice send an impossibility if you want to keep the sends after the inserts.

**Comment by mkruselj** [comment]
I suppose that would then be the end of this issue, then? @PopeKehoeIII should we close it?

**Comment by magnetophon** [comment]
Why close it?

Looks like it's not going to be implemented in the near future, but it should still be possible, IIUC.
It's only impossible under certain conditions, as @PopeKehoeIII said:
- if the FX routing is not made significantly more flexible
- if you want to keep the sends after the inserts

**Comment by baconpaul** [comment]
Right I agree lets not close it but it is hard; basically we need to have a pre- and post- inline fx send and only pre can be per voice. That's @PopeKehoeIII observation and it adds just more switching and optionality to the requirement. But not impossible. (for instance you do it without any switching by implementing two sliders: a voice-send and a scene-send and have you able to use both. Scene send acts as today and voice-send sends pre-inline fx voice modulated amount to the same fx that scene send sends to.).

**Comment by mkruselj** [comment]
Having two different types of send sliders for the same exact send would be confusing. It's probably better to just have a right-click option on the send level param to switch it to Pre-FX mode, which would automatically be per voice (no need to have it at scene level IMHO).

**Comment by baconpaul** [comment]
yeah sure
kinda the same thing though logically (although two sliders lets you blend pre and post I guess).

**Comment by mkruselj** [comment]
I guess it is easier to implement as separate parameters, yeah. Mockup:

![image](https://github.com/user-attachments/assets/b855f71c-4b8a-423a-8eab-d30c64b84051)

**Comment by mkruselj** [comment]
Closing this issue in favor of #1010, which predates it and also subsumes it (see https://github.com/surge-synthesizer/surge/issues/1010#issuecomment-4311531380 in particular).
